hrm-domain/hrm-core/profile/profileService.ts

```diff
@@ -187,14 +261,21 @@ export const associateProfileToProfileFlag = async (
     }
 
     return await db.task(async t => {
-      (
-        await profileDB.associateProfileToProfileFlag(t)(
-          accountSid,
-          profileId,
-          profileFlagId,
-          validUntil,
-        )
-      ).unwrap(); // unwrap the result to bubble error up (if any)
+      await Promise.all([
+        profileDB
```

Comment:
Should this be a sequential transaction rather than attempted in parallel? I think audit updates should be in transactions with the data changes if our audit data is to be considered reliable.

Also, there probably isn't much point in doing `Promise.all` on multiple queries reusing the same `task`, since there won't be a performance benefit. A task represents work done on a single underlying database connection, and since db connections are serial, synchronous channels the queries will need to be executed sequentially anyway, regardless on the async JS being used. All that using Promise.all really achieves is adding an element of unpredictability to the order in which the queries will be executed

Reply:
This makes sense, I'm so used to `Promise.all`ify everything that didn't even though if it made sense here hehe. Thanks for the feedback!